test/e2e/memcached_test.go

```diff
@@ -479,23 +479,36 @@ func memcachedScaleTest(t *testing.T, f *framework.Framework, ctx *framework.Tes
 		t.Fatalf("Failed to unmarshal memcached CR: (%v)", err)
 	}
 	obj.SetNamespace(namespace)
-	err = f.Client.Get(context.TODO(), types.NamespacedName{Name: obj.GetName(), Namespace: obj.GetNamespace()}, &obj)
-	if err != nil {
-		return fmt.Errorf("failed to get memcached object: %s", err)
-	}
-	// update memcached CR size to 4
-	spec, ok := obj.Object["spec"].(map[string]interface{})
-	if !ok {
-		return errors.New("memcached object missing spec field")
-	}
-	spec["size"] = 4
-	err = f.Client.Update(context.TODO(), &obj)
+
+	err = wait.Poll(retryInterval, timeout, func() (done bool, err error) {
+		err = f.Client.Get(context.TODO(), types.NamespacedName{Name: obj.GetName(), Namespace: obj.GetNamespace()}, &obj)
+		if err != nil {
+			return false, fmt.Errorf("failed to get memcached object: %s", err)
+		}
+		// update memcached CR size to 4
+		spec, ok := obj.Object["spec"].(map[string]interface{})
+		if !ok {
+			return false, errors.New("memcached object missing spec field")
+		}
+		spec["size"] = 4
+		err = f.Client.Update(context.TODO(), &obj)
+		if err != nil {
+			if apierrors.IsConflict(err) {
+				return false, nil
+			}
+			return false, err
+		}
+		return true, nil
+	})
 	if err != nil {
-		return err
+		return fmt.Errorf("could not update memcached CR: %v", err)
 	}
 
 	// wait for example-memcached to reach 4 replicas
-	return e2eutil.WaitForDeployment(t, f.KubeClient, namespace, "example-memcached", 4, retryInterval, timeout)
+	if err := e2eutil.WaitForDeployment(t, f.KubeClient, namespace, "example-memcached", 4, retryInterval, timeout); err != nil {
+		return fmt.Errorf("failed waiting for 4 example-memcached replicas: %v", err)
```

Comment:
Nit: Not related to this PR but can we make the deployment sizes `3` and `4` test variables instead of using them as they are.
This message and others can become outdated if we tweak this test in the future.

Reply:
Same for the test-framework tests below.